### DISCUSSION TITLE
Shortcut features

### DISCUSSION THREAD

**Discussion by Bouh** [4ian/GDevelop#2348] [Enhancement ideas (and their technical discussions)]
Hi here some bugs with the shortcut features.
@NilayMajorwar 

I have a azerty keyboard, because i'am french, and my language of windows is set on french too.
I can't use the shortcut M for toggle the mask on the scene.
I need use the comma ,

If i wish setup the M key in the preferences i can't.
[See in video the probleme](https://streamable.com/2vzz59), you will se in the bottom right of my screen the inputs.

Can we have a better arrow ?, the icon arrow on left is pretty , but not between texts :/
![image](https://user-images.githubusercontent.com/1670670/91617379-3909f780-e988-11ea-9b5a-9d2c85e32225.png)

The L shortcut for Layers, and others modal windows in the editor are not toggleable with the shortcut. Can we do something for use the toggable function when the shortcut are pressed ?

Warning for the instance list the searchbar is focused automatically maybe you can remove it for make the shortcut toggleable.

## Other details
* Win10, FR
* Curent master branch (after release of b98 and before b99)

**Comment by Bouh** [comment]
Can we also remove the gap between the searbar and the list ?
This was done for others list like the autocomplete and the list of objects in the instructions editor.

**Comment by 4ian** [comment]
> Can we have a better arrow ?, the icon arrow on left is pretty , but not between texts :/

Yeah I've been advising to use such an arrow, but the result is not that good. Actually, we can rework it and also maybe remove the arrows on the left:
![image](https://user-images.githubusercontent.com/1280130/91635276-f6c7d100-e9ee-11ea-8d7f-9c16f7d5cc35.png)

In VSCode, there is no such "icon" for each command: 
![image](https://user-images.githubusercontent.com/1280130/91635282-06471a00-e9ef-11ea-937e-757e48e6773c.png)

> Warning for the instance list the searchbar is focused automatically maybe you can remove it for make the shortcut toggleable.

On the other hand, it's important to have the focus on this for the workflow (being able to start typing the name of the object you're searching, then pressing Enter to select it). 
This being said, I agree with the toggling it's something to test. Maybe that could better work with a shortcut like `Alt+something` - with a modifier key so that it could be used even when a text field is selected?

**Comment by Silver-Streak** [comment]
The ALT key may be a bad choice, just because then you become limited by whatever other ALT commands you have in the App.(Within Windows, at least)

Maybe Shift+Key or ALT+Shift+Key?

**Comment by nilaymaj** [comment]
Hey, sorry for the delay!

> I can't use the shortcut M for toggle the mask on the scene.

If it's still not working, can you go to [keycode.info](https://keycode.info) to check the key code (particularly `event.code`) of the `M` key on your keyboard? That should hopefully indicate where the issue comes from.

> the icon arrow on left is pretty , but not between texts

Yeah, that looks worse than desired :sweat_smile: - I'll remove the arrow in the text, sure. About the icon on the left, I don't know - it looks good to me but then VSCode looks good to me too! Perhaps I can first check how each option looks with MUI.

> toggle-able shortcuts

Yeah, the current non-toggle shortcuts are not very intuitive and break a probably quicker workflow for dealing with panes, but as @4ian said, the auto-focus searchbars are an exception. Maybe we can make it toggle-able for panes that don't have auto-search?

**Comment by AlexandreSi** [comment]
@Bouh is this issue still a thing?

**Reply by Bouh** [reply]
Yes still.
I need use the comma ,
Instead of M as setup in preferences with a french azerty keyboard.
